crates/ruff_python_parser/src/lexer.rs

```diff
@@ -428,10 +429,46 @@ impl<'source> Lexer<'source> {
                     self.cursor.bump();
                     value.push('\\');
                 }
+                '?' => {
+                    self.cursor.bump();
+                    let mut question_count: u8 = 1;
+                    while self.cursor.eat_char('?') {
+                        question_count += 1;
+                    }
+
+                    // The help end magic command is valid if it has 1 or 2 question marks.
+                    // If the `value` is empty i.e., we are at the start of the line,
+                    // then we can't be sure if it is a help end magic command or not.
+                    if question_count > 2
```

Comment:
If I understand this correctly, then it is about lexing out all starting question marks? 

Would it make sense to add the logic for eating all leading quotes at the top of `lex_magical_command`?

```rust
self.cursor.eat_while(|c| c == '?');
```

We could even move it to the `lex_magical_command`'s call-site where we matched the `?` if we only need to eat over th leading `?`when the start token is a `?`. So the question is, do we also need to eat over the `?` for `%??abc??` or is this invalid?

Reply:
> If I understand this correctly, then it is about lexing out all starting question marks?

No, this is about lexing the _ending_ question marks if present. The terminology is ["help end"](https://github.com/ipython/ipython/blob/292e3a23459ca965b8c1bfe2c3707044c510209a/IPython/core/inputtransformer2.py#L465) magic command to avoid confusion but if you think this is unclear we can rename it?.

> Would it make sense to add the logic for eating all leading quotes at the top of `lex_magical_command`?

I don't think so because we only need to consume it if the magic command has `?` tokens (either 1 or 2) at the end. We won't know this until the cursor is at the end.